torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1858,6 +1871,35 @@ def _lazy_init(self) -> None:
         for handle in self._handles:
             self._init_param_attributes(handle)
         self._exec_order_data.init(self, self.process_group)
+        # Initialize inter-node process group for hybrid shard comm. We assume
+        # passed in process group is the process group over which the model is
```

Comment:
> We assume passed in process group is the process group over which the model is sharded

Does "passed in process group" refer to the one passed to the FSDP constructor?

If so, I am wondering if this is really an assumption. It seems to be deterministically true given how FSDP sharding is implemented, and there would be a bug with FSDP if it somehow used a different process group that is not the one passed into the constructor.